### ISSUE TITLE
[PW-6549] Error in virtuel orders and credit cart in 8.2.0

### ISSUE THREAD

**Issue by dronerdk** [Adyen/adyen-magento2#1445]
**Describe the bug**
When we try to checkout with a virtuel order/product it is not possible to select credit card payment due to a js bug

**To Reproduce**
1) Add virtuel product to cart
2) Go to checkout
3) Logo and text is missing for credit card payment and it is not possible to select and checkout
4) JS errors in console

**Magento version**
2.4.3-p1

**Plugin version**
8.2.0

**Screenshots**
<img width="2220" alt="Skærmbillede 2022-04-09 kl  18 50 39" src="https://user-images.githubusercontent.com/15999222/162583706-753000aa-614b-4a94-afc2-8882d134cfd5.png">

**Comment by adamj88** [comment]
They have an icon referenced where the JS allows this to be undefined, yet the check in the template for it doesn't allow no value to be presented, giving the error in the screenshot above.

The change which allows icon to be passed as undefined can be seen here: https://github.com/Adyen/adyen-magento2/commit/dd8758885b028d599e84440d879b5e9e1cea34fc#diff-1db253bbc5ff55b3b5e8821c73ac41509277d4c8d262bfad9862f50a2d5ec88eR100-R102

With it being accessed at this point in the frontend template and producing the error here: https://github.com/Adyen/adyen-magento2/compare/8.1.1...8.2.0#diff-a90f2bcde1042beac4226c992249534b91c937e5b3843c2214cedb7023102403R32

To work around this we have modified the KO template to access the variable a little differently allowing it to be passed as undefined:

```diff
--- view/frontend/web/template/payment/cc-form.html
+++ view/frontend/web/template/payment/cc-form.html
@@ -29,7 +29,7 @@
                data-bind="attr: {'id': getCode()}, value: getCode(), checked: isChecked, click: selectPaymentMethod, visible: isRadioButtonVisible()"/>
         <label data-bind="attr: {'for': getCode()}" class="label">

-            <!-- ko if: icon -->
+            <!-- ko if: $data.icon -->
             <img data-bind="attr: {
                 'src': icon.url,
                 'alt': getTitle() + ' logo'
```

**Comment by tnaber** [comment]
Hi @dronerdk,

Thank you opening this issue. We will be investigating this.
@adamj88, thank you for the proposed fix! Would you be able to open a PR for you solution?

Cheers,
Titus

**Comment by dronerdk** [comment]
Thanks @adamj88 and @tnaber. The patch works in magento developer mode. If we set the site in production mode we fill face same problem again. Any fix for that?
Screenshot from site in production mode with cc-form.html patch applied:
<img width="2052" alt="Skærmbillede 2022-04-25 kl  08 52 41" src="https://user-images.githubusercontent.com/15999222/165036102-e783e3f0-132c-4179-a277-8f19596a476a.png">

**Comment by dronerdk** [comment]
We still have this error after the patch.
Are there some other things we can look into?

**Comment by tnaber** [comment]
Hi @dronerdk,

On my side the fix also didn't fully resolve the issue. As seen in your screenshot it is most likely part of the larger JS issue indicated in #1440. I've have opened a draft PR #1456 to implement a general solution so you can follow the progress over there, the aim is to also resolve this issue in that PR.

Cheers,
Titus

**Comment by dronerdk** [comment]
Ok thanks @tnaber . When you know, please tell me a date for a fix. We will roll back the version.

**Comment by tnaber** [comment]
Hi @dronerdk,

Feel free to test the updated fix in #1456, if it is sufficient it should be ready for 8.2.3

Cheers,
Titus
